lib/src/rules/parameter_assignments.dart

```diff
@@ -124,75 +106,95 @@ class ParameterAssignments extends LintRule {
   }
 }
 
-class _Visitor extends SimpleAstVisitor<void> {
+class _DeclarationVisitor extends RecursiveAstVisitor {
```

Comment:
Do we want to locate assignments within nested functions? For example:
```
void f(int p) {
  void g() {
    p = 3;
  }
  g();
}
```
If so, we should have a test for that case. If not, you'll need to override `visitFunctionExpression` to stop the recursive visitor from going inside.

Reply:
Yep.  Test added.

Thanks!